packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -108,18 +108,32 @@ - (int64_t)onPlayerSetup:(FVPVideoPlayer *)player {
 
   int64_t playerIdentifier;
   if (textureBasedPlayer) {
-    playerIdentifier = [self.registry registerTexture:(FVPTextureBasedVideoPlayer *)player];
+    playerIdentifier = [self.registry registerTexture:textureBasedPlayer];
     [textureBasedPlayer setTextureIdentifier:playerIdentifier];
   } else {
     playerIdentifier = self.nextNonTexturePlayerIdentifier--;
   }
 
+  NSObject<FlutterBinaryMessenger> *messenger = self.messenger;
+  NSString *channelSuffix = [NSString stringWithFormat:@"%lld", playerIdentifier];
+  // Set up the player-specific API handler, and its onDispose unregistration.
+  SetUpFVPVideoPlayerInstanceApiWithSuffix(messenger, player, channelSuffix);
+  __weak typeof(self) weakSelf = self;
+  BOOL isTextureBased = textureBasedPlayer != nil;
+  player.onDisposed = ^() {
+    SetUpFVPVideoPlayerInstanceApiWithSuffix(messenger, nil, channelSuffix);
```

Comment:
The addition of `onDisposed` to the base class is to allow easily cleaning up the communication channel, without having to push all the communication channel wire-up into the player class (which would be a bigger change, and is also undesirable for potential conversion to using FFI to talk directly to the player instance).

Reply:
If the structure of the native code is that all the channel-related details are contained entirely within the main plugin class (this file), and the player classes are just receiving calls, then switching to FFI can be done with almost no changes to the player instance classes.

In contrast, if I had pushed instance channel setup in the player instance classes in this PR, then a future change to FFI would require more changes in those classes to unwind that, instead of mostly changing this plugin-structure-specific class.